File: PODAAC/TP_GPN_2PfP003_002_19921013_003126_19921013_012737.nc.py
"""

This example code illustrates how to access and visualize a PO.DAAC
 TOPEX_POSEIDON_GDR_F L2 netCDF-4 file in Python.

If you have any questions, suggestions, or comments on this example, please use
the HDF-EOS Forum (http://hdfeos.org/forums).  If you would like to see an
example of any other NASA HDF/HDF-EOS data product that is not listed in the
HDF-EOS Comprehensive Examples page (http://hdfeos.org/zoo), feel free to
contact us at eoshelp@hdfgroup.org or post it at the HDF-EOS Forum
(http://hdfeos.org/forums).

Usage: save this script and run

    $python TP_GPN_2PfP003_002_19921013_003126_19921013_012737.nc.py

Tested under: Python 3.9.13 :: Miniconda (x86_64)
Last Update: 2023-11-09
"""

import matplotlib as mpl
import matplotlib.pyplot as plt
import numpy as np
from mpl_toolkits.basemap import Basemap
from netCDF4 import Dataset

# Open netCDF-4 file.
FILE_NAME = "TP_GPN_2PfP003_002_19921013_003126_19921013_012737.nc"
nc = Dataset(FILE_NAME)

# Read dataset.
DATAFIELD_NAME = "swh_ku"
dset = nc.variables[DATAFIELD_NAME]
data = dset[:]
latitude = nc.variables["latitude"][:]
longitude = nc.variables["longitude"][:]

print(longitude)
latmin = np.min(latitude)
latmax = np.max(latitude)
lonmin = np.min(longitude)
lonmax = np.max(longitude)

m = Basemap(
    projection="cyl",
    resolution="l",
    llcrnrlat=latmin, urcrnrlat = latmax,
    llcrnrlon=lonmin, urcrnrlon = lonmax
)
m.drawcoastlines(linewidth=0.5)
m.drawparallels(np.arange(-90, 91, 45), labels=[True, False, False, False])
m.drawmeridians(np.arange(-180, 180, 45), labels=[False, False, False, True])
m.scatter(longitude, latitude, c=data, s=10, cmap=plt.cm.jet,
          edgecolors=None, linewidth=0)
cb = m.colorbar()
units = dset.units
cb.set_label("Unit: " + units)
long_name = dset.long_name
plt.title("{0}\n {1}".format(FILE_NAME, long_name))
fig = plt.gcf()

# Save plot.
pngfile = "{0}.py.png".format(FILE_NAME)
fig.savefig(pngfile)

File: PODAAC/TP_GPN_2PfP003_002_19921013_003126_19921013_012737_nc.m
%
%  This example code illustrates how to access and visualize
% PO.DAAC TOPEX_POSEIDON_GDR_F L2 netCDF-4 file in MATLAB. 
%
%  If you have any questions, suggestions, comments  on this example, 
% pleaseuse the HDF-EOS Forum (http://hdfeos.org/forums). 
%
%  If you would like to see an  example of any other NASA
% HDF/HDF-EOS data product that is not listed in the
% HDF-EOS Comprehensive Examples page (http://hdfeos.org/zoo), 
% feel free to contact us at eoshelp@hdfgroup.org or post it at the
% HDF-EOS Forum (http://hdfeos.org/forums).
%
% Usage:save this script and run (without .m at the end)
%
%  $matlab -nosplash -nodesktop -r TP_GPN_2PfP003_002_19921013_003126_19921013_012737_nc
%
% Tested under: MATLAB R2023b
% Last updated: 2023-11-09

% Open the netCDF-4/HDF5 file.
FILE_NAME='TP_GPN_2PfP003_002_19921013_003126_19921013_012737.nc';

% List file content.
% ncdisp(FILE_NAME)

ncid = netcdf.open(FILE_NAME, 'nowrite');

% Read data from a data field.
DATAFIELD_NAME='/swh_ku';

data = ncread(FILE_NAME, DATAFIELD_NAME);

% Read units.
units = ncreadatt(FILE_NAME, DATAFIELD_NAME, "units");

% Read long_name.
long_name = ncreadatt(FILE_NAME, DATAFIELD_NAME, "long_name");

% Read latitude data.
DATAFIELD_NAME='/latitude';
lat = ncread(FILE_NAME, DATAFIELD_NAME);
lat = double(lat);
                 
% Read longitude data.
DATAFIELD_NAME='/longitude';
lon = ncread(FILE_NAME, DATAFIELD_NAME);
lon = double(lon);

% Close the file.
netcdf.close(ncid);

% Create the graphics figure.
f=figure('Name', FILE_NAME, ...
	 'Renderer', 'zbuffer', ...
	 'Position', [0,0,800,600], ...
	 'visible', 'off');
axesm ('eqdcylin', 'Frame', 'on', 'Grid', 'on', ...
      'MeridianLabel', 'on', 'ParallelLabel','on', ...
      'MLabelParallel', 'south')

scatterm(lat, lon, 1, data);
colormap('Jet');

coast = load('coastlines.mat');
plotm(coast.coastlat,coast.coastlon,'k')

% Change the value if you want to have more than 10 tick marks.
h = colorbar();
set (get(h, 'title'), 'string', units)
tightmap;

title({FILE_NAME;long_name}, 'Interpreter', 'None', ...
      'FontSize',10,'FontWeight','bold');

saveas(f, [FILE_NAME  '.m.png']);
exit
